### PR TITLE
Bootstrap the repo

### PR DESCRIPTION
**Pull request by bladyjoker**
Hey @gnumonik, we have some goodies here I want you to check out:

1. Testing works locally, not yet in CI (noisy Spago)
2. Added code quality and assistance tools
3. Nix
4. CI
5. ToData/FromData is still untested

### CODE REVIEW

File: src/Language/PureScript/Bridge/SumType.hs [hunk 1]
```diff
@@ -109,17 +144,19 @@ data Instance (lang :: Language)
   | Enum
   | Bounded
   | HasConstrIndex
+  | ToData
```
Comment (gnumonik) [inline src/Language/PureScript/Bridge/SumType.hs:147]: Do we need this? I.e. do we really expect anyone using the library to want to generate HasConstrIndices instances but not To/FromData instances? 

Either way I think `mkSumTypeIndexed` should generate `ToData/FromData` instances by default. Maybe we should rename it to `mkPlutusSum` or something like that?

**Comment by gnumonik** [comment]
I think everything here is good. Going to merge it in a second.

File: src/Language/PureScript/Bridge/SumType.hs (hunk 1 above)
Comment (bladyjoker) [reply src/Language/PureScript/Bridge/SumType.hs:147]: I completely agree!

**Comment by bladyjoker** [comment]
@gnumonik that's it from me today, I didn't manage to generate the instance clauses so if you can, I'd be happy if you could continue with the following:

- Turn the test green `1) buildBridge without lens-code-gen tests the generation of a CTL HasConstrIndices/ToData/FromData` 
- Figure out flakiness of `writePSTypesWith should produce aeson-compatible argonaut instances` if doable
- Merge this PR
- Introduce the (the essential) changes you have pending in your existing PRs:
  - your comments and suggestions from here
  - everything needed to actually generate HasConstrIndices, ToData, FromData
- Prepare a PR in Cardax to use the new version

We need to be able to continue user registration tomorrow, for that we need ToData/FromData instances.

Cheers,
Draz

**Comment by gnumonik** [comment]
Notes (mainly for my own benefit): 
- This should _finally_ work the way it needs to for Cardax. We now generate ToData/FromData instances. 

- I'm punting on the Argonaut/spago2nix stuff for right now. The existing tests say the instances are argonaut compatible and I'm not _that_ sure how to hook up spago2nix output to our flake. I'll come back to this later today if I have time. 

- I think these changes are sufficient for us to integrate into Cardax. Actually merging now.

**Pull request merged**